software/controller/lib/core/system_constants.h

```diff
@@ -0,0 +1,120 @@
+/* Copyright 2021, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// All constants related to the ventilator design should live here. These
+// include pin mappings, venturi geometry, ...
+
+// Reference abbreviations ([RM], [PCB], etc) are defined in hal/README.md
+
+#pragma once
+
+#include "gpio.h"
+#include "units.h"
+
+/**************************************************************************************
+ *                                ELECTRICAL CONSTANTS                                *
+ **************************************************************************************/
+
+// The analog sensor on the daughterboard is hooked up to PC0 (see [PCB]) aka ADC1_IN1 (see [DS])
+static constexpr GPIO::AdcChannel InterimBoardAnalogPressure{GPIO::Port::C, 0, 1};
+
+// The O2 sensor on the daughterboardis hooked up to PC3 (see [PCB]), aka ADC1_IN2 (see [DS])
+static constexpr GPIO::AdcChannel InterimBoardOxygenSensor{GPIO::Port::C, 3, 2};
+
+// "U4 Inhale Flow" sensor on the PCB is pin PA4 (see [PCB]), aka ADC1_IN9 (see [DS])
+static constexpr GPIO::AdcChannel U4InhaleFlow{GPIO::Port::A, 4, 9};
+
+// "U3 Patient Pressure" sensor on the PCB, i.e PA1 (see [PCB]), aka ADC1_IN6 (see [DS])
+static constexpr GPIO::AdcChannel U3PatientPressure{GPIO::Port::A, 1, 6};
+
+// "U5 Exhale Flow" sensor on the PCB is pin PB0 (see [PCB]), aka ADC1_IN15 (see [DS])
+static constexpr GPIO::AdcChannel U5ExhaleFlow{GPIO::Port::B, 0, 15};
+
+// Blower is hooked up to PB3 (see [PCB]),
+// which can be linked to Timer2 chanel 2 using AF1 (see [DS], p77)
+static constexpr GPIO::PwmChannel BlowerChannel = {GPIO::Port::B, 3, GPIO::AlternativeFunction::AF1,
```

Comment:
don't think this requires assignment operator `=`

Reply:
indeed